### PR TITLE
using json.net for all serialization due to the use of primitive list…

### PR DESCRIPTION
**Pull request by ekolve**
…s (SceneBounds.cornerPoints), which get ignored by JsonUtility

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def push_build(build_archive_name, archive_sha256, include_private_scenes):
 def _local_build_path(prefix="local"):
     if platform.system() == "Darwin":
         suffix = "OSXIntel64.app"
-        build_path = "unity/builds/thor-{}-{}/Contents/MacOS/thor-local-OSXIntel64".format(
+        build_path = "unity/builds/thor-{}-{}/Contents/MacOS/AI2-Thor".format(
```
Comment (AlvaroHG) [inline tasks.py:59]: Does this break anything?

File: ai2thor/tests/data/metadata-schema.json [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+        "objectIdsInBox",
+        "objects",
+        "reachablePositions",
+        "sceneBounds",
+        "sceneName",
+        "screenHeight",
+        "screenWidth",
+        "segmentedObjectIds",
+        "thirdPartyCameras",
+        "visibleRange"
```
Comment (AlvaroHG) [inline ai2thor/tests/data/metadata-schema.json:664]: nice!

File: unity/Assets/Scripts/AgentManager.cs [hunk 3]
```diff
@@ -1382,3 +1383,28 @@ public class TypedVariable {
     public string type;
     public object value;
 }
+
+public class ShouldSerializeContractResolver : DefaultContractResolver
+{
+   public static readonly ShouldSerializeContractResolver Instance = new ShouldSerializeContractResolver();
+
+   protected override JsonProperty CreateProperty( MemberInfo member,
```
Comment (AlvaroHG) [inline unity/Assets/Scripts/AgentManager.cs:1391]: Hopefully this does not make it much slower. Why did we not need this before, or did we? Is this to the net framework change and support with a particular Newtonsoft version?

File: ai2thor/tests/test_server.py [hunk 4]
```diff
@@ -13,12 +13,10 @@ def generate_multi_agent_form(metadata, sequence_id=1):
     agent2['agentId'] = 1
     agent1 = metadata
     agents = [agent1, agent2]
-    action_returns = [None, None]
     boundary = b'--OVCo05I3SVXLPeTvCgJjHl1EOleL4u9TDx5raRVt'
     data = b'\r\n' + boundary + b'\r\nContent-Type: text/plain; charset="utf-8"\r\nContent-disposition: form-data; name="metadata"\r\n\r\n'
     data += json.dumps(dict(agents=agents, sequenceId=sequence_id, activeAgentId=1)).encode('utf8')
     data += b'\r\n' + boundary + b'\r\nContent-Type: text/plain; charset="utf-8"\r\nContent-disposition: form-data; name="actionReturns"\r\n\r\n'
-    data += json.dumps(action_returns).encode('utf8')
     data += b'\r\n' + boundary + b'\r\nContent-Type: text/plain; charset="utf-8"\r\nContent-disposition: form-data; name="token"\r\n\r\n'
```
Comment (AlvaroHG) [inline ai2thor/tests/test_server.py:20]: do we still have an `actionReturn` value per agent? why is this different?

**Comment by AlvaroHG** [comment]
A couple of questions, but looks as expected LGTM

File: tasks.py (hunk 1 above)
Comment (ekolve) [reply tasks.py:59]: it was broken before - the binary name changed with the update to 2019

**Pull request merged**